File: Java/Web-Wicket/src/main/java/com/untitleddoc/cadencecalc/web/wicket/WicketApplication.java
package com.untitleddoc.cadencecalc.web.wicket;

import jakarta.persistence.EntityManagerFactory;
import lombok.extern.slf4j.Slf4j;
import org.apache.wicket.RuntimeConfigurationType;
import org.apache.wicket.markup.html.WebPage;
import org.apache.wicket.protocol.http.WebApplication;

/**
 * Application object for your web application. If you want to run this
 * application without deploying, run the Start class.
 *
 * @see com.untitleddoc.web.wicket.Start#main(String[])
 */
@Slf4j
public final class WicketApplication extends WebApplication {

	/**
	 * @return デフォルトページクラス
	 * @see org.apache.wicket.Application#getHomePage()
	 */
	@Override
	public Class<? extends WebPage> getHomePage() {
		return HomePage.class;
	}

	/**
	 * @see org.apache.wicket.Application#init()
	 */
	@Override
	public void init() {
		super.init();

		//レスポンスのエンコーディング設定
		getRequestCycleSettings().setResponseRequestEncoding("UTF-8");
	}

	@Override
	public RuntimeConfigurationType getConfigurationType() {
		return super.getConfigurationType();
	}
}

File: Web-Wicket-Kotlin/src/main/kotlin/com/untitleddoc/cadencecalc/web/wicket/kotlin/WicketApplication.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.untitleddoc.cadencecalc.web.wicket.kotlin

import org.apache.wicket.Page
import org.apache.wicket.RuntimeConfigurationType
import org.apache.wicket.protocol.http.WebApplication

/*

  @author kisiy
  Created on 2018/02/12
*/

class WicketApplication : WebApplication() {
    /**
     * @see org.apache.wicket.Application#init()
     */
    override fun init(): Unit {
        super.init()

		//レスポンスのエンコーディング設定
		getRequestCycleSettings().setResponseRequestEncoding("UTF-8")
    }

    override fun getHomePage(): Class<out Page> {
        return HomePage::class.java
    }

    override fun getConfigurationType() : RuntimeConfigurationType {
        return super.getConfigurationType()
    }
}